core/src/main/python/synapse/ml/llm/HuggingFaceCausallmTransform.py

```diff
@@ -0,0 +1,373 @@
+        self.config.update(kwargs)
+
+
+def camel_to_snake(text):
+    return re.sub(r"(?<!^)(?=[A-Z])", "_", text).lower()
+
+
+class HuggingFaceCausalLM(
+    Transformer, HasInputCol, HasOutputCol, DefaultParamsReadable, DefaultParamsWritable
+):
+
+    modelName = Param(
+        Params._dummy(),
+        "modelName",
+        "model name",
+        typeConverter=TypeConverters.toString,
+    )
+    inputCol = Param(
+        Params._dummy(),
+        "inputCol",
+        "input column",
+        typeConverter=TypeConverters.toString,
+    )
+    outputCol = Param(
+        Params._dummy(),
+        "outputCol",
+        "output column",
+        typeConverter=TypeConverters.toString,
+    )
+    modelParam = Param(Params._dummy(), "modelParam", "Model Parameters")
+    modelConfig = Param(Params._dummy(), "modelConfig", "Model configuration")
+    useFabricLakehouse = Param(
+        Params._dummy(),
+        "useFabricLakehouse",
+        "Use FabricLakehouse",
+        typeConverter=TypeConverters.toBoolean,
+    )
+    lakehousePath = Param(
+        Params._dummy(),
+        "lakehousePath",
+        "Fabric Lakehouse Path for Model",
+        typeConverter=TypeConverters.toString,
+    )
+    deviceMap = Param(
+        Params._dummy(),
+        "deviceMap",
+        "device map",
+        typeConverter=TypeConverters.toString,
+    )
+    torchDtype = Param(
+        Params._dummy(),
+        "torchDtype",
+        "torch dtype",
+        typeConverter=TypeConverters.toString,
+    )
+
+    @keyword_only
+    def __init__(
+        self,
+        modelName=None,
+        inputCol=None,
+        outputCol=None,
+        useFabricLakehouse=False,
+        lakehousePath=None,
+        deviceMap=None,
+        torchDtype=None,
+    ):
+        super(HuggingFaceCausalLM, self).__init__()
+        self._setDefault(
+            modelName=modelName,
+            inputCol=inputCol,
+            outputCol=outputCol,
+            modelParam=ModelParam(),
+            modelConfig=ModelConfig(),
+            useFabricLakehouse=useFabricLakehouse,
+            lakehousePath=None,
+            deviceMap=None,
+            torchDtype=None,
+        )
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    def load_model(self):
+        """
+        Loads model and tokenizer either from Fabric Lakehouse or the HuggingFace Hub,
+        depending on the 'useFabricLakehouse' param.
+        """
+        model_name = self.getModelName()
+        model_config = self.getModelConfig().get_config()
+        device_map = self.getDeviceMap()
+        torch_dtype = self.getTorchDtype()
+
+        if device_map:
+            model_config["device_map"] = device_map
+        if torch_dtype:
+            model_config["torch_dtype"] = torch_dtype
+
+        if self.getUseFabricLakehouse():
+            local_path = (
+                self.getLakehousePath() or f"/lakehouse/default/Files/{model_name}"
+            )
```

Comment:
switch to just use cachePath and then in our docs well say this is a good place to store things